### PR TITLE
Chocolatey install: add proxy support

### PR DESCRIPTION
**Pull request by dougireton**
This adds support for installing Chocolatey from behind a proxy.
Inspired by http://powershell.com/cs/blogs/tips/archive/2010/11/12/get-webclient-with-proxy-authentication.aspx

Also added test-kitchen support for Win2012r2 and 2008R2

### CODE REVIEW

**Comment by ferventcoder** [comment]
@mwrock @guilhem Thoughts on this?

File: recipes/default.rb [hunk 1]
```diff
@@ -32,11 +32,13 @@
   cmd = 'cmd.exe'
 end
 
+chocolatey_install_script = "$wc = New-Object Net.WebClient; $wc.UseDefaultCredentials = $true; $wc.Proxy.Credentials = $wc.Credentials; $wc.DownloadString('#{node['chocolatey']['Uri']}')"
```
Comment (mwrock) [inline recipes/default.rb]: I'd consider this:
```
command = <<-EOS
  $wc = New-Object Net.WebClient
  $wp=[system.net.WebProxy]::GetDefaultProxy()
  $wp.UseDefaultCredentials=$true
  $wc.Proxy=$wp
  Invoke-Expression ($wc.DownloadString('#{node['chocolatey']['Uri']}'))
EOS

encoded_script = command.encode('UTF-16LE', 'UTF-8')
chocolatey_install_script = Base64.strict_encode64(encoded_script)

File: recipes/default.rb [hunk 2]
```diff
@@ -32,11 +32,13 @@
...
+
 batch 'install chocolatey' do
   architecture arch
   interpreter cmd
   code <<-EOH
-    powershell -noprofile -inputformat none -noninteractive -executionpolicy bypass -command "iex ((new-object net.webclient).DownloadString('#{node['chocolatey']['Uri']}'))"
+    powershell -noprofile -inputformat none -noninteractive -executionpolicy bypass -command "#{chocolatey_install_script} | Invoke-Expression"
```
Comment (mwrock) [inline recipes/default.rb]: Now you would make this:
```
powershell -noprofile `
                  -inputformat none `
                  -noninteractive `
                  -executionpolicy bypass `
                  -EncodedCommand #{chocolatey_install_script}

File: recipes/default.rb (hunk 1 above)
Comment (mwrock) [inline recipes/default.rb]: This is more readabl I think and the command is slightly different. I no longer have a proxy I can test against but I have used this for years without issue. That said, if @dougireton 's script works both behind a proxy and without proxy, I'm fine with that.

**Comment by mwrock** [comment]
In addition to the above notes +100 for the test-kitchen stuff!

**Comment by dougireton** [comment]
@mwrock I tested this with and without a proxy and it works in both cases.

**Comment by dougireton** [comment]
I like the suggested changes. I'll resubmit today.

**Comment by dougireton** [comment]
Thanks for your work on this cookbook. We are using it at Nordstrom in our open-source [chefdk_bootstrap](https://supermarket.chef.io/cookbooks/chefdk_bootstrap) cookbook.

**Comment by mwrock** [comment]
Looks good and kitchen converge worked with a working choco install

**Comment by dougireton** [comment]
Can you please release a new version to Public Supermarket?

**Comment by mwrock** [comment]
I dont have collaborator rights to the supermarket account. Deferring to @guilhem and @ferventcoder.

**Comment by ferventcoder** [comment]
I thought @guilhem hooked you up with rights? Guess that still needs to happen.

**Pull request merged**